projects/libs/core-library-typescript/src/block-storage/block-storage.ts

```diff
@@ -131,4 +136,14 @@ export class BlockStorage {
   private async putBlock(block: types.Block): Promise<void> {
     await this.db.put<string>(block.header.height.toString(), JSON.stringify(block));
   }
+
+  public async startupCheck(): Promise<StartupStatus> {
+
+    if (!this.transactionPool) {
+      return <StartupStatus>{ name: this.SERVICE_NAME, status: STARTUP_STATUS.FAIL };
+    }
+
+    const lastBlock = await this.getLastBlock();
```

Comment:
Needs to be in try/catch because if `getLastBlock` fails we want to return an error.

Reply:
Added